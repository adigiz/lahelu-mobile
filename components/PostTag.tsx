import { Text, TouchableOpacity, View } from "react-native";

interface PostTagProps {
  tag: string;
}

const getTagColor = (tag: string): string => {
  switch (tag.toLowerCase()) {
    case 'sawer':
      return 'bg-orange-300';
    default:
      return '';
  }
};

const getBorderColor = (tag: string): string => {
  switch (tag.toLowerCase()) {
    case 'sawer':
      return 'border-orange-300';
    default:
      return 'border-gray-300';
  }
};

export default function PostTag({ tag }: PostTagProps) {
  return (
    <TouchableOpacity activeOpacity={0.7}>
      <View
        className={`overflow-hidden rounded-full border px-[10] py-[3] font-semibold ${getBorderColor(
          tag
        )} ${getTagColor(tag)}`}
      >
        <Text>{`# ${tag}`}</Text>
      </View>
    </TouchableOpacity>
  );
}
