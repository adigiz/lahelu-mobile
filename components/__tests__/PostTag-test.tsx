import { render } from '@testing-library/react-native';
import React from 'react';

import PostTag from '@/components/PostTag';

describe('PostTag', () => {
    it('should matches the snapshot with the "sawer" tag', () => {
        const { toJSON } = render(<PostTag tag="sawer" />);

        expect(toJSON()).toMatchSnapshot();
    });

    it('should matches the snapshot with a generic tag', () => {
        const { toJSON } = render(<PostTag tag="generic" />);
        
        expect(toJSON()).toMatchSnapshot();
    });
});
