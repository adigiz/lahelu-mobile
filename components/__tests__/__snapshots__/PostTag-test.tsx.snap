// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTag should matches the snapshot with a generic tag 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    className="overflow-hidden rounded-full border px-[10] py-[3] font-semibold border-gray-300 "
  >
    <Text>
      # generic
    </Text>
  </View>
</View>
`;

exports[`PostTag should matches the snapshot with the "sawer" tag 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    className="overflow-hidden rounded-full border px-[10] py-[3] font-semibold border-orange-300 bg-orange-300"
  >
    <Text>
      # sawer
    </Text>
  </View>
</View>
`;
